lib/castle/context/default.rb

```diff
@@ -4,36 +4,58 @@ module Castle
   module Context
     class Default
       def initialize(request, cookies = nil)
-        @client_id = Extractors::ClientId.new(request, cookies || request.cookies).call
-        @headers = Extractors::Headers.new(request).call
-        @request_ip = Extractors::IP.new(request).call
+        @pre_headers = HeaderFilter.new(request).call
+        @cookies = cookies || request.cookies
+        @request = request
       end
 
       def call
-        defaults.merge!(additional_defaults)
-      end
-
-      private
-
-      def defaults
         {
-          client_id: @client_id,
+          client_id: client_id,
           active: true,
           origin: 'web',
-          headers: @headers,
-          ip: @request_ip,
+          headers: headers,
+          ip: ip,
           library: {
             name: 'castle-rb',
             version: Castle::VERSION
           }
-        }
+        }.tap do |result|
+          result[:locale] = locale if locale
+          result[:user_agent] = user_agent if user_agent
+        end
       end
 
-      def additional_defaults
-        {}.tap do |result|
-          result[:locale] = @headers['Accept-Language'] if @headers['Accept-Language']
-          result[:user_agent] = @headers['User-Agent'] if @headers['User-Agent']
-        end
+      private
+
+      # locale
```

Comment:
let's remove comments that just state the method name, they don't make any sense. If you care about the linter, having `# @return [String]` should be enough